syntax = "proto3";
package protos;

enum VectorType {
  BIAS = 0;
  LEFT = 1;  // left = query
  RIGHT = 2; // right = index
}

message OptimConfig {
  float l1 = 1;
  float l2 = 2;
  float alpha = 3;
  float beta = 4;
  float emb_alpha = 5;
  float emb_beta = 6;
  float emb_l1 = 7;
  float emb_l2 = 8;
  uint32 emb_size = 9;
}

message FeatureConfig {
  uint64 slot_id = 1;
  string name = 2;
  VectorType vec_type = 3;
  int32 cross = 4;
}

message AllConfig{
  OptimConfig optim_config = 1;
  repeated FeatureConfig feature_list = 2;
  bool is_feature_signed = 3;
  bool group_sparse = 4;

  repeated string train_list = 5;
  repeated string predict_list = 6;
  string save_path = 7;
  string load_path = 8;
}